### ISSUE TITLE
Fully implement Signal handling

### ISSUE THREAD

**Issue by kjzl** [WSSMRKS/42_minishell#71]
Problems:
- Early exit command leads to double prompt
```
ms/kwurster@c4a11c3:~/projects/minishell$ sleep 5
[DBG] parsed AST: sleep 5
Simple Command:
├─ Heredoc flag: 0
├─ Words:
│  ├─ 'sleep'
│  ├─ '5'
├─ Argv:
│  ├─ [0]: 'sleep'
│  ├─ [1]: '5'
└─ Redirections:
^C
ms/kwurster@c4a11c3:~/projects/minishell$ ms/kwurster@c4a11c3:~/projects/minishell$ 
```
- Minishell in Minishell
- Heredoc can not be canceled with SIGINT

**Comment by kjzl** [comment]
Maybe this is interesting?
https://github.com/bminor/bash/blob/6794b5478f660256a1023712b5fc169196ed0a22/execute_cmd.c#L5662

**Comment by WSSMRKS** [comment]
use ioctl(stdout, tiocti, "") to "press enter without pressing enter"
$? when executing signal --> 128 + Number of signal
Signalhandlers in stephanies minishell:
- interactive mode
- heredoc
- execution mode
- pipeloop (we won't be implementing this)

**Comment by WSSMRKS** [comment]
closed with #96
